Storage/src/SigningHelper.php

```diff
@@ -882,4 +896,46 @@ private function buildQueryString(array $input)
 
         return implode('&', $q);
     }
+
+    /**
+     * Retry logic for signBlob
+     *
+     * @param callable $signBlobFn  A callable that perform the actual signBlob operation.
+     * @param string $resourceName The resource name for logging or retry strategy determination.
+     * @param array $args Arguments for the operations, include preconditions
+     * @return string The signature genarated by signBlob.
+     * @throws ServiceException If non-retryable error occur.
+     * @throws \RuntimeException If retries are exhausted.
+     */
+    private function retrySignBlob(callable $signBlobFn, string $resourceName = 'signBlob', array $args = [])
+    {
+        $attempts = 0;
+        $maxRetries = 5;
+        $invocationId = Uuid::uuid4()->toString();
+        $args['retryStrategy'] = StorageClient::RETRY_ALWAYS;
+
+        // Generate a retry decider function using the RetryTrait logic.
+        $retryDecider = $this->getRestRetryFunction($resourceName, 'execute', $args);
```

Comment:
My understanding is idempontentOps lists are sourced from https://github.com/googleapis/google-cloud-php/blob/main/Storage/src/Connection/Rest.php and are mapped with https://github.com/googleapis/google-cloud-php/blob/main/Storage/src/Connection/ServiceDefinition/storage-v1.json. However, the 'signBlob' function seems to be implemented in the core library. Please let me know if I'm mistaken.

Reply:
Thanks for the suggestion!  Adding signBlob to the idempotencyOps list is a great idea.  This will simplify the logic in this function and make it more maintainable. I've implemented the change based on your recommendation.

Reply:
You got it; sorry i didn't clarify further.